core/src/main/java/ai/timefold/solver/core/impl/score/director/ValueRangeManager.java

```diff
@@ -436,26 +438,41 @@ public ReachableValues getReachableValues(GenuineVariableDescriptor<Solution_> v
                                         entityDescriptor.getEntityClass().getSimpleName(),
                                         variableDescriptor.getVariableName(), valuesSize));
             }
-            var reachableValuesMap = new IdentityHashMap<Object, ReachableItemValue>((int) valuesSize);
+            Class<?> valueClass = null;
```

Comment:
This whole nested logic is incredibly long. Invert it - return `values` if not null, then deal with the other case without nesting.